### PR TITLE
Add tests for duplicate dims in batch_dims and input_dims

### PR DESCRIPTION
**Pull request by maxrjones**
**Description of proposed changes**

This PR adds two tests for BatchGenerator when there are one or more dimensions included in both input_dims and batch_dims. The testing utilities are updated to account for this case. The tests currently fail due to https://github.com/xarray-contrib/xbatcher/issues/131, but including them will be helpful for finishing #132.

### CODE REVIEW

File: xbatcher/tests/test_generators.py [hunk 1]
```diff
@@ -115,6 +115,21 @@ def test_batch_1d_concat(sample_ds_1d, input_size):
...
+    """
+    Test batch generation for a 1D dataset using ``concat_input_dims`` when
+    the same dimension occurs in ``input_dims`` and `batch_dims``
+    """
+    bg = BatchGenerator(
+        sample_ds_1d, input_dims={"x": 5}, batch_dims={"x": 10}, concat_input_dims=True
+    )
+    validate_generator_length(bg)
+    expected_dims = get_batch_dimensions(bg)
+    for n, ds_batch in enumerate(bg):
```
Comment (andersy005) [inline xbatcher/tests/test_generators.py:128]: Remove unused `enumarate()` since the index `n` isn't used 

```suggestion
    for ds_batch in bg:
```

File: xbatcher/tests/test_generators.py [hunk 2]
```diff
@@ -109,12 +109,27 @@ def test_batch_1d_concat(sample_ds_1d, input_size):
     )
     validate_generator_length(bg)
     expected_dims = get_batch_dimensions(bg)
-    for n, ds_batch in enumerate(bg):
+    for ds_batch in bg:
         assert isinstance(ds_batch, xr.Dataset)
         validate_batch_dimensions(expected_dims=expected_dims, batch=ds_batch)
         assert "x" in ds_batch.coords
```
Comment (norlandrhagen) [inline xbatcher/tests/test_generators.py:117]:
```suggestion
@pytest.mark.xfail
```

File: xbatcher/tests/test_generators.py [hunk 3]
```diff
@@ -218,6 +233,23 @@ def test_batch_3d_1d_input_batch_dims(sample_ds_3d, concat):
         validate_batch_dimensions(expected_dims=expected_dims, batch=ds_batch)
```
Comment (norlandrhagen) [inline xbatcher/tests/test_generators.py:235]:
```suggestion
test_batch_3d_1d_input_batch_dims
```

**Review by norlandrhagen — APPROVED** [review]
Maybe add some xfails so CI passes?

**Comment by maxrjones** [comment]
> Maybe add some xfails so CI passes?

Good idea, thanks!
